django/db/models/base.py

```diff
@@ -572,6 +572,11 @@ def __getstate__(self):
         """Hook to allow choosing the attributes to pickle."""
         state = self.__dict__.copy()
         state['_state'] = copy.copy(state['_state'])
+        state['_memoryview_attrs'] = []
+        for k, v in list(state.items()):
+            if isinstance(v, memoryview):
+                state['_memoryview_attrs'].append((k, bytes(v)))
+                state.pop(k)
```

Comment:
This is a bit hacky (but backward compatible), so an alternative patches and suggestions are welcome!

Reply:
I learned a bit from this about memoryviews. Thank you!

It might not be worth it, since it's premature optimization, but if you are in the mood, here are a few ideas:

1. Assign the copied bytes to a local variable list and only add them to the state if the list is non-empty. This saves a dictionary entry if no memoryviews exist, and it saves a hash lookup into the state for each memoryview.

2. Pop entries from the state after creating the local variable list. This would allow iterating over state's items without copying them into a list.